mutations/core.py

```diff
@@ -131,3 +131,21 @@ def validate(cls, raise_on_error=False, **kwargs):
             else:
                 return ValidationResult(is_valid=False, errors=error_dict)
         return ValidationResult(is_valid=is_valid, errors=error_dict)
+
+
+@six.add_metaclass(MutationBase)
+class AsyncMutation(Mutation):
+    @classmethod
+    async def run(cls, raise_on_error=False, **kwargs):
+        """Validate the inputs and then awaits execute() to run the command. """
+        instance = cls(cls.__name__, inputs = kwargs)
+        is_valid, error_dict = instance._validate()
+
+        if not is_valid:
+            if raise_on_error:
+                raise error.MutationFailedValidationError(error_dict)
+            else:
+                return Result(success=False, return_value=None, errors=error_dict)
+
+        result = await instance.execute()
```

Comment:
the only difference between `Mutation` and `AsyncMutation` is that we await the `execute()` function

Reply:
Nice, this looks good. One question I've been thinking about and curious to hear your take.. what if we were to do it like this (pseudocode):

```python
import asyncio
import inspect

...

class Mutation():
    def run(cls, raise_on_error=False, loop=None, **kwargs):
        if inspect.isawaitable(...):
            result = asyncio.run(...)
        else:
            ...
```

I think the way you're proposing is more obvious and probably a better developer experience. What do you think?

Reply:
Oh I like that!! it'd be nice to have just one class that automatically handles sync vs async invocations. Will update my PR :) thanks for the suggestion!

Reply:
@omarish updated! tests pass as before